aws-android-sdk-iot/src/main/java/com/amazonaws/mobileconnectors/iot/AWSIotSslUtility.java

```diff
@@ -56,7 +84,20 @@ public static SSLSocketFactory getSocketFactoryWithKeyStore(KeyStore keyStore)
             throws NoSuchAlgorithmException, UnrecoverableKeyException, KeyStoreException,
             KeyManagementException {
 
-        SSLContext context = SSLContext.getInstance("TLSv1.2");
+        SSLContext context;
+
+        if (portNumber == 443) {
+            try {
+                // Attempt to use Conscrypt
+                Security.addProvider(new OpenSSLProvider());
+                context = SSLContext.getInstance(TLS_V_1_2, "Conscrypt");
+            } catch (NoSuchProviderException e) {
+                // Fallback to system SSLContext
```

Comment:
can you add a log statement saying that the provider does not exist and we are falling back to default provider?